File: ProductProtoPortal/ClientApp/src/app/model/AdditionalCost.ts
export interface AdditionalCost{
  id:number;
  description:string;
  comment:string;
  cost:number;
  additionalCostTemplate?:AdditionalCostTemplate;
  additionalCostTemplateId:number;
  workId:number;
  subPost:string;
}
export interface AdditionalCostTemplate{
  id:number;
  name:string;
  canPost:boolean;
  canItem:boolean;

}

File: ProtoLib2/Model/AdditionalCost.cs
using System.ComponentModel.DataAnnotations;

namespace ProtoLib2.Model;

public class AdditionalCost
{
    public long Id { get; set; }

    [MaxLength(128)] public string Description { get; set; }

    [MaxLength(128)] public string Comment { get; set; }

    public decimal Cost { get; set; }

    public long WorkId { get; set; }
    public Work Work { get; set; }


    public long AdditionalCostTemplateId { get; set; }
    public AdditionalCostTemplate AdditionalCostTemplate { get; set; }
}